### ISSUE TITLE
How to resume an archive collection after Academic limit has been reached?

### ISSUE THREAD

**Issue by Michael-Gauthier** [DocNow/twarc#526]
Hello everyone,

I'm sorry if I'm asking a dumb question, or if the answer has already been provided somewhere around here. I've looked around and couldn't find one, but I'm not a computer scientist, so I may have missed it.

I'm an academic, I have access to the Academic track of Twitter's API and I'm about to start a study on the discourse around global warming. My main concern is thus to collect tweets containing keywords such as 'global warming', 'climate change' etc that have been emitted since 2010... 

However, the limit being set at 10M tweets collected per month, I know that my corpus is going to take more than one month to be full, and I would like to know how, after the collection is going to stop after the first 10M tweets are collected, I can resume the process where it left off without collecting the same tweets again. 

In other words, how can I start a new collection from the archive and carry on the data collection where it previously stopped?

Again, I'm originally a linguist, not a programmer, so apologies if my question is not clear, but any indication to help me with that would be very much appreciated, so thanks in advance! : )

**Comment by SamHames** [comment]
Some suggestions (assuming you're using the command line):

1. You can use the --start-time and --end-time options to limit your collection to a specific timerange, and maybe consider collecting 1 year/ 3months/ 1 month of data at a time. 
2. If you have a partial collection, you can use the --since_id/--until_id options to restart a collection at the last tweet you collected (retrieving the tweet id from the collected JSON).
3. You mentioned you're a linguist - depending on your analysis it might be appropriate to exclude retweets from a collection by adding "-is:retweet" to your query, since they're just a verbatim copy of the original tweet. The tweet statistics already includes the number of times it has been retweeted, so you may not need the actual retweets themselves.
4. Whatever you're doing, you can use the `counts` command to explore how big the collection is before you even start, which will help you plan.

I'm happy to expand on any of those if you have any questions.

**Comment by Michael-Gauthier** [comment]
Hello Sam,

Thanks a lot for your quick and thorough answer it helps a lot!

Yes, I do use the command line, sorry for not mentioning that earlier.

I did think about limiting the collection to short periods of time only, as it is what seems to be the easiest way around, but I was wondering whether there was a more automated way of doing that, although this solution suits me perfectly. 

Indeed I originally thought of not including retweets, precisely because of the point you raised, however I just had a discussion with a colleague, who suggested that it might be interesting in the case of retweets followed by a personal note or comment, as in the following imaginary example:

'RT: "blablabla" ==> Very true, I hadn't thought of that'

But the question is, in that case, would the whole tweet be considered as a retweet in the same way a verbatim retweet would?

I did not know about the `counts` command, thank you!

Thanks again for your help, I used Twitter for research a few years ago, but since then things have changed a bit and having some help to get back into it is really nice! : )

**Comment by SamHames** [comment]
Most retweets are native or button retweets, these are more like links to
the original tweet than verbatim quotes - the manual RT is rarely used. I'm
on my phone right now so can't look it up easily, but there might be some
operators you can use to distinguish between them. From a vague memory
there might be an "is:nativeretweet" operator, but I'm not sure.

**Comment by edsu** [comment]
That is some excellent advice from @SamHames. Personally I would start by using the `counts` command to generate a CSV of daily counts and then plan out a series of commands to run each month.

    twarc2 counts '"global warming" OR "climate change"' --archive --csv --granularity day counts.csv

I just ran it (note without `is:retweet`) and the daily count summed to 140,623,218 tweets. That would take over a year to collect (unless you collaborated with some other researchers with Academic access). So you might want to see what the counts look like with `is:retweet` if those aren't important for your research.

![fig](https://user-images.githubusercontent.com/33829/131490190-b336e63d-ff6a-41e5-a28d-7a308a6d7401.png)

I wonder if the peak in 2019 is Greta!?

**Comment by edsu** [comment]
Please feel free to reopen if you need additional help!

**Comment by edsu** [comment]
That's pretty interesting you got 111M without retweets. I would've expected it to be less considering the count with retweets is 140M. Did you end up using the same query otherwise? `'"global warming" OR "climate change"'`?

**Comment by Michael-Gauthier** [comment]
@edsu it is interesting indeed! And I too, would have expected a far lower number of tweets without the RTs...

I did use the same query in both cases, although my filters were slightly different from yours (I added a couple more keywords, and I only started the query in 2010). In the end, I have 122M matches with the RTs, and 111M without, which seems very surprising... Here is the command I used to get the results (with or without the "-is:retweet" bit of course).

`twarc2 counts '"global warming" OR "globalwarming" OR "climate change" OR "climatechange" -is:retweet' --archive --start-time 2010-01-01 --end-time 2020-01-01 --csv --granularity day counts.csv`

I'll have to explore the data further to check for potential inconsistencies or mistakes on my end, but maybe there aren't just so many RTs when dealing with this topic...

**Comment by igorbrigadir** [comment]
Oh, before I forget, might still be worth adding a feature that reads the last ID collected from an existing file and picks up where it left off like @SamHames suggested in 2. https://github.com/DocNow/twarc/issues/526#issuecomment-908940500

**Comment by Michael-Gauthier** [comment]
Thanks a lot for the suggestion @igorbrigadir ! I indeed thought about doing that at first, as it seems to be the most efficient and automated method to carry out a data collection, however while doing tests to check how twarc works and what the end files look like, I realized that for the exact same collection parameters with a `--limit `set at 1000:

1) The order the tweets are added into the files varies from one collection to the other (still while starting the exact same collection)

2) All the tweets are not exactly the same (some tweets are present in certain output files, others are missing, although the vast majority is indeed similar)

So I was wondering whether a collection method relying on providing the latest ID collected would be reliable... Are archived tweets collected in a systematic way based on the collection parameters provided (e.g. chronological order)? That would explain why using the last ID of a previous collection would work, but that would not explain the disparities observed when doing several identical sample collections several times in a row...

Such a method would indeed be ideal, but any indication as to how the requests are processed through the API would be really helpful! : )

NOTE: I realize that this is off topic with my original question, so if I would be better off creating new threads, or unarchiving this one, please let me know @edsu (or the others of course), I'm not really super comfortable using Github, so I may not be aware of the expected courses of action...

**Comment by SamHames** [comment]
1. For the is:retweet filter, I think you need to group the first set of operators, because of the precedence rules, I think it only applies to the last operator in the chain of ORs:
> twarc2 counts '("global warming" OR "globalwarming" OR "climate change" OR "climatechange") -is:retweet' --archive --start-time 2010-01-01 --end-time 2020-01-01 --csv --granularity day counts.csv

Note the brackets grouping all the keywords. This only matches ~46 million tweets, which is what I think we expected.

2. I'm not quite sure I understand what you're saying about order - how are you checking that? If you're not seeing reverse chronological order you might need to check your approach. Note that the tweet_id embeds a timestamp of when the tweet is created, so that's baked into the platform.

**Comment by Michael-Gauthier** [comment]
@edsu Hi, I'm coming back to you to ask you what probably is a silly question, but I'd be curious to know how you managed to do that visualization of the number of tweets emitted thanks to the output file of `twarc2 counts`? I now have additional data, and a more refined search pattern, and I've been trying to reproduce the same graph but without much success so far, so I'm turning to you... Are you using R to do that? 

Thanks in advance for any help you could provide. : )

**Comment by edsu** [comment]
@Michael-Gauthier you probably want to use the `--csv` option? Then it's real easy to visualize. I used Python & plotly, but I'm sure there is an equivalent (and probably more attractive) way of doing it in R. You can see an example of doing the data collection and vis (first with matplotlib and then plotly) in this notebook if you want:

https://colab.research.google.com/drive/1d8x7JCG-DQ5WCAXfdz23Y2zdwqzC85zk#scrollTo=Ufr2D0C1ry-N
